### PR TITLE
Feature/svcs 2428 default rql builder.double encode values to true and mark double encode values as deprecated

### PR DESCRIPTION
**Pull request by chad-pelly**

### CODE REVIEW

File: src/rql/types.ts [hunk 1]
```diff
@@ -13,7 +13,28 @@ export type RQLBuilderOptions = {
...
+   * **Previous behaviour:**
+   * - RQL values could be provided as double encoded to the RQL builder
+   * - `rqlBuilder.doubleEncodeValues` could be set to `true` to automatically double encode RQL values globally
+   * - rqlBuilder({ doubleEncodeValues: true })... would automatically double encode RQL values per builder
+   *
+   *
+   * **New behaviour:**
+   * - RQL values are automatically double encoded when using the RQL builder
+   *
+   * **How to transition to 8.0.0:**
```
Comment (chad-pelly) [inline src/rql/types.ts:28]: transition -> migrate?

File: src/rql/types.ts [hunk 2]
```diff
@@ -13,7 +13,28 @@ export type RQLBuilderOptions = {
...
+   * - Update any RQL values to no longer be encoded e.g.
+   * - old value: 'Hypertension%2520%252D%2520STAGE%25201'
+   * - new value: 'Hypertension - STAGE 1'
+   *
+   * **Alternatively and not advised**:
```
Comment (AaronTrippaers) [inline src/rql/types.ts:33]: I am not sure I would call not advisable. I might be the easy and most gradual way to migrate for customers to just `rqlBuilder.doubleEncodeValues`  to `false` and then set it to true on a per request basis when they are sure that request still works?

**Review by AaronTrippaers — COMMENTED** [review]
One tiny comment

File: src/rql/types.ts [hunk 3]
```diff
@@ -13,7 +13,28 @@ export type RQLBuilderOptions = {
...
 
 export interface RqlBuilderFactory {
-  // Used to set the global value to double encode RQL values using the syntax `rqlBuilder.doubleEncodeValues` (Default true).
+  /**
+   * @deprecated this property is only meant to be used for backwards compatibility when upgrading to v8.0.0
+   *
+   * **Previous behaviour:**
+   * - RQL values could be provided as double encoded to the RQL builder
+   * - `rqlBuilder.doubleEncodeValues` could be set to `true` to automatically double encode RQL values globally
+   * - rqlBuilder({ doubleEncodeValues: true })... would automatically double encode RQL values per builder
```
Comment (wwaaijer-exh) [inline src/rql/types.ts:22]:
```suggestion
   * - `rqlBuilder({ doubleEncodeValues: true })...` would automatically double encode RQL values per builder
```

File: src/rql/types.ts [hunk 4]
```diff
@@ -13,7 +13,28 @@ export type RQLBuilderOptions = {
...
+   * - RQL values are automatically double encoded when using the RQL builder
+   *
+   * **How to transition to 8.0.0:**
+   * - Update any RQL values to no longer be encoded e.g.
+   * - old value: 'Hypertension%2520%252D%2520STAGE%25201'
+   * - new value: 'Hypertension - STAGE 1'
+   *
+   * **Alternatively and not advised**:
+   * - `rqlBuilder.doubleEncodeValues` can be set to `false` to disable double encoding RQL values globally
+   * - rqlBuilder({ doubleEncodeValues: false })... would disable automatically double encoding RQL values per builder
```
Comment (wwaaijer-exh) [inline src/rql/types.ts:35]:
```suggestion
   * - `rqlBuilder({ doubleEncodeValues: false })...` would disable automatically double encoding RQL values per builder
```

**Review by wwaaijer-exh — COMMENTED** [review]
Looking good 👌

**Review by AaronTrippaers — APPROVED** [review]

**Review by wwaaijer-exh — APPROVED** [review]
👌

**Pull request merged**